pkg/daemon/config_drift_monitor.go

```diff
@@ -43,9 +49,8 @@ type ConfigDriftMonitorOpts struct {
 	OnDrift func(error)
 	// The currently applied MachineConfig.
 	MachineConfig *mcfgv1.MachineConfig
-	// The Systemd dropin path location.
-	// Defaults to /etc/systemd/system
-	SystemdPath string
+	// System's root path
+	RootPath string
```

Comment:
suggestion: I think we should do some kind of validation on this value. Here's why:

Let's say that this gets called with a value like, "hello". When we add this onto the paths, we get something like, e.g., `hello/etc/systemd/system`. At the very least, we should validate that the value provided here starts with a `/` or is not empty.

Reply:
In the case when we are running tests the root dir becomes the `/tmp/xxxx/`
In the non-test case it would make sense to pass `/` as the rootPath and use `filepath.join` as suggested before for appending systemd path as mention before.
Yes agreed we should run validation checks